### PR TITLE
[openmpi] Support RBAC

### LINKED ISSUE #718: openmpi controller exited with error
@jiezhang Could you help me on this?

```
controller entered
controller exited
Traceback (most recent call last):
  File "/root/controller/main.py", line 17, in <module>
    main()
  File "/root/controller/main.py", line 13, in main
    ctl.wait_master_terminated()
  File "/root/controller/controller.py", line 44, in wait_master_terminated
    phase = self._get_master_phase()
  File "/usr/local/lib/python3.6/site-packages/retrying.py", line 49, in wrapped_f
    return Retrying(*dargs, **dkw).call(f, *args, **kw)
  File "/usr/local/lib/python3.6/site-packages/retrying.py", line 212, in call
    raise attempt.get()
  File "/usr/local/lib/python3.6/site-packages/retrying.py", line 247, in get
    six.reraise(self.value[0], self.value[1], self.value[2])
  File "/usr/local/lib/python3.6/site-packages/six.py", line 693, in reraise
    raise value
  File "/usr/local/lib/python3.6/site-packages/retrying.py", line 200, in call
    attempt = Attempt(fn(*args, **kwargs), attempt_number, False)
  File "/root/controller/controller.py", line 55, in _get_master_phase
    pod = self.api.read_namespaced_pod(self.master, self.namespace)
  File "/usr/local/lib/python3.6/site-packages/kubernetes/client/apis/core_v1_api.py", line 18163, in read_namespaced_pod
    (data) = self.read_namespaced_pod_with_http_info(name, namespace, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/kubernetes/client/apis/core_v1_api.py", line 18254, in read_namespaced_pod_with_http_info
    collection_formats=collection_formats)
  File "/usr/local/lib/python3.6/site-packages/kubernetes/client/api_client.py", line 321, in call_api
    _return_http_data_only, collection_formats, _preload_content, _request_timeout)
  File "/usr/local/lib/python3.6/site-packages/kubernetes/client/api_client.py", line 155, in __call_api
    _request_timeout=_request_timeout)
  File "/usr/local/lib/python3.6/site-packages/kubernetes/client/api_client.py", line 342, in request
    headers=headers)
  File "/usr/local/lib/python3.6/site-packages/kubernetes/client/rest.py", line 231, in GET
    query_params=query_params)
  File "/usr/local/lib/python3.6/site-packages/kubernetes/client/rest.py", line 222, in request
    raise ApiException(http_resp=r)
kubernetes.client.rest.ApiException: (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'Content-Type': 'application/json', 'X-Content-Type-Options': 'nosniff', 'Date': 'Wed, 25 Apr 2018 05:36:04 GMT', 'Content-Length': '305'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods \"slim-377c8-master\" is forbidden: User \"system:serviceaccount:tfworkflow:default\" cannot get pods in the namespace \"tfworkflow\"","reason":"Forbidden","details":{"name":"slim-377c8-master","kind":"pods"},"code":403}

```

### PR DESCRIPTION
**Pull request by jiezhang**
Fix #718

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/756)

### CODE REVIEW

File: kubeflow/openmpi/service.libsonnet [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
   all(params):: [
     $.service(params),
+    $.serviceAccount(params),
+    $.clusterRoleBinding(params),
   ],
 
   name(params):: params.name,
```
Comment (ankushagarwal) [inline kubeflow/openmpi/service.libsonnet]: To simplify, can you delete this and replace `$.name(params)` with `params.name` throughout the file

**Comment by jiezhang** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[kubeflow/openmpi/service.libsonnet, line 8 at r1](https://reviewable.io/reviews/kubeflow/kubeflow/756#-LBb_z9v_gzp4GVtxG0R-r1-8:-LBbf3rk-087zkZnEDMX:bunrhxd) ([raw file](https://github.com/kubeflow/kubeflow/blob/3d18883d16c8bdd11fcf69013c773122c2e04a34/kubeflow/openmpi/service.libsonnet#L8)):*
<details><summary><i>Previously, ankushagarwal (Ankush Agarwal) wrote…</i></summary><blockquote>

To simplify, can you delete this and replace `$.name(params)` with `params.name` throughout the file
</blockquote></details>

No, I can't do that. It's referenced in workloads.jsonnet.

```
serviceAccountName: service.name(params)
subdomain: service.name(params)

```

---


*Comments from [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/756)*

**Pull request merged**